### PR TITLE
JAMES-2847 Removing Memory and MailRepository DeletedMessageVault

### PR DESCRIPTION
**Pull request by chibenwa**
Issues:

 - Route test (purge) needed to be corrected:
   - purge need clock management
   - task type was updated  hence tests where no longer relevant

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project james-server-memory-guice: There was a timeout or other error in the fork -> [Help 1]
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] 
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] 
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] For more information about the errors and possible solutions, please read the following articles:
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] 
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] After correcting the problems, you can resume the build with the command
[a677caa83943605d4684eb73577c764486f8356d] [ERROR]   mvn <goals> -rf :james-server-memory-guice
```
test this please

**Comment by chibenwa** [comment]
```
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project james-server-memory-guice: There was a timeout or other error in the fork -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
```
[a677caa83943605d4684eb73577c764486f8356d] [INFO] Apache James :: Server :: Memory - guice injection . FAILURE [  02:02 h]
[a677caa83943605d4684eb73577c764486f8356d] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project james-server-memory-guice: There was a timeout or other error in the fork -> [Help 1]
```

**Comment by chibenwa** [comment]
:green_apple: before CI reboot

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[b9c14008055543a7c87005f5bc92e63e6441dbb6] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
[b9c14008055543a7c87005f5bc92e63e6441dbb6] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
```
test this please

**Comment by chibenwa** [comment]
```
[b9c14008055543a7c87005f5bc92e63e6441dbb6] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[b9c14008055543a7c87005f5bc92e63e6441dbb6] LastClientMsg: c SEARCH FLAGGED
[b9c14008055543a7c87005f5bc92e63e6441dbb6] Expected: '\* SEARCH 1 2 3 4'
[b9c14008055543a7c87005f5bc92e63e6441dbb6] Actual   : '* SEARCH'
```

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java [hunk 1]
```diff
@@ -72,7 +70,8 @@
     private final RetentionConfiguration retentionConfiguration;
 
     @Inject
-    BlobStoreDeletedMessageVault(MetricFactory metricFactory, DeletedMessageMetadataVault messageMetadataVault,
+    @VisibleForTesting
```
Comment (blackheaven) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java]: `public` & `@VisibleForTesting`?

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**